src/libOpenImageIO/imagebufalgo_opencv.cpp

```diff
@@ -136,7 +137,182 @@ IplImage *
 ImageBufAlgo::to_IplImage (const ImageBuf &src)
 {
 #ifdef USE_OPENCV
-    return NULL;
+    ImageSpec spec = src.spec();
+    ImageBuf tmp = src;
```

Comment:
It seems like the "usual" case (3 or 4 channel image) is going to perform this copy (more on that in a moment) unnecessarily, since just below (if spec.channels >= 3) it will call ImageBufAlgo::channels that also will copy the image into tmp.

But now that I'm looking at the code, I'm concerned that ImageBuf=ImageBuf is not well behaved. I've intentionally tried to define `ImageBuf::operator=` to be unimplemented so that this is disallowed. Yet somehow it compiles. It's not supposed to.

OK, so the way I think this should really work is like this:

```
    const ImageBuf *tmp = &src;
    ImageBuf localcopy;
    if (spec.channels >= 3) {
        ...
        ImageBufAlgo::channels(localcopy, src, ...);
        tmp = &localcopy;
        ...
    }

```

then some uses of `tmp` will need to be replaced by `*tmp` below, but you can see that there are never unnecessary copies made.